### DISCUSSION TITLE
Display rows spanning several terminal lines and allowing newlines in cell content

### DISCUSSION THREAD

**Discussion by 1a7r0ch3** [andmarti1424/sc-im#534] [Ideas]
Currently when displaying a table in sc-im, each row occupies only one terminal line.

It would be interesting to have a mechanism allowing a cell content to be displayed over several lines; of course its entire row must be adjusted in height accordingly.

In particular, when a cell content is too long to fit within its column width, it is currently either displayed with stars "***" to indicate that it cannot fit, or its content overlaps the cells on its rights if the `overlap` option is set.

By allowing rows with height spanning over several terminal lines, one can imagine to wrap around long cell content.

Illustration :

    ┌───────────────┬─────────────┐
    │ short content │*************│
    ├───────────────┼─────────────┤
    │ short content │long content overlapping next cell
    ├───────────────┼─────────────┤
    │               │long content │
    │ short content │wrapping     │
    │               │around       │
    └───────────────┴─────────────┘

Moreover, this would allow (with careful escaping and parsing rules) to deal with cell contents actually containing newlines ; see for instance issue #410.

Now I have no idea how hard it would be to achieve this. Also some design decision must be made, for instance regarding what to do with the vertical position of "short content" (I guess similar mechanism than horizontal justification can be used, defaulting to vertical centering), or what to do in case of `autojus` of column width (I guess an option setting a "maximum autoadjust width" could be used).

**Comment by andmarti1424** [comment]
Excellent @1a7r0ch3. thank you!

**Reply by andmarti1424** [reply]
@1a7r0ch3 with my latest commit, rows now can have height higher than 1 line. The wrapping is still pending.

**Reply by 1a7r0ch3** [reply]
Sounds great! But is there any way to see the feature in action?

**Reply by andmarti1424** [reply]
just do `fj` to increase height in a cell that has long content

**Reply by 1a7r0ch3** [reply]
Great! So you did implement some wrapping also. Well done!

**Reply by andmarti1424** [reply]
still pending the autowrap feature. it shouldnt be that difficult.. 
currently working on multiple frozen rows/cols..
